//
// PROJECT:         Aspia
// FILE:            category/category_printer.proto
// LICENSE:         Mozilla Public License Version 2.0
// PROGRAMMERS:     Dmitry Chapyshev (dmitry@aspia.ru)
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package aspia.proto;

message Printers
{
    message Item
    {
        string name            = 1;
        bool is_default        = 2;
        bool is_shared         = 3;
        string share_name      = 4;
        string port_name       = 5;
        string driver_name     = 6;
        string device_name     = 7;
        string print_processor = 8;
        string data_type       = 9;
        string server_name     = 10;
        string location        = 11;
        string comment         = 12;
        int32 jobs_count       = 13;
        int32 paper_width      = 14;
        int32 paper_length     = 15;
        int32 print_quality    = 16;

        enum Orientation
        {
            ORIENTATION_UNKNOWN   = 0;
            ORIENTATION_PORTRAIT  = 1;
            ORIENTATION_LANDSCAPE = 2;
        }

        Orientation orientation = 17;
    }

    repeated Item item = 1;
}
